src/WebJobs.Extensions.DocumentDB/Trigger/DocumentDBTriggerListener.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor;
+    using Microsoft.Azure.Documents.Client;
+    using Microsoft.Azure.WebJobs.Host.Executors;
+    using Microsoft.Azure.WebJobs.Host.Listeners;
+
+    internal class DocumentDBTriggerListener : IListener, IChangeFeedObserverFactory
+    {
+        private readonly ITriggeredFunctionExecutor executor;
+        private readonly ChangeFeedEventHost host;
+        public DocumentDBTriggerListener(ITriggeredFunctionExecutor executor, DocumentCollectionInfo documentCollectionLocation, DocumentCollectionInfo leaseCollectionLocation, ChangeFeedHostOptions leaseHostOptions)
+        {
+            this.executor = executor;
+            string hostName = Guid.NewGuid().ToString();
+            this.host = new ChangeFeedEventHost(hostName, documentCollectionLocation, leaseCollectionLocation, new ChangeFeedOptions(), leaseHostOptions);
+        }
+
+        public void Cancel()
+        {
+            this.StopAsync(CancellationToken.None).Wait();
+        }
+
+        public IChangeFeedObserver CreateObserver()
+        {
+            return new DocumentDBTriggerObserver(this.executor);
+        }
+
+        public void Dispose()
+        {
+            //Nothing to dispose
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            return host.RegisterObserverFactoryAsync(this);
```

Comment:
A couple of clarifications now that I'm thinking through this (we may have discussed this already):

1. Will there only ever be one instance that is hooked up to this trigger? I can't remember what we said -- or was it one per partition?
2. Is there a maximum per-batch document count that you can configure?
3. Are these observers executed synchronously? For (a silly) example, if I have `Task.Delay(60000)` in my function code so it takes a while for me to execute each batch -- will my observer be called again on a separate thread while the previous one is waiting? Or will it wait until the previous is done?